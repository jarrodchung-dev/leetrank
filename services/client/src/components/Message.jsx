import React, { useState, useEffect } from "react";

const Message = (props) => {
  const [visible, setVisible] = useState(true);
  useEffect(() => {
    setTimeout(() => {
      setVisible(false);
    }, 2000);
  });
  return (
    <>
      {visible && (
        <div className="container is-fluid has-text-centered" id="notification">
          <div className={`notification is-${props.type}`}>
            <button className="delete" onClick={() => setVisible(!visible)}></button>
            <span>{props.text}</span>
          </div>
        </div>
      )}
    </>
  );
};

export default Message;
