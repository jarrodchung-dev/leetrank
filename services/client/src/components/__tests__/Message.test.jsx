import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Message from "../Message.jsx";

const messages = [
  {
    text: "Successful message",
    type: "success"
  },
  {
    text: "Danger message",
    type: "danger"
  }
];

messages.forEach((message) => {
  describe(`<Message ${message.type} />`, () => {
    it("renders with the correct text and type", () => {
      const wrapper = shallow(<Message text={message.text} type={message.type} />);
      const element = wrapper.find(`.notification.is-${message.type}`);
      expect(element.length).toBe(1);
      const span = wrapper.find("span");
      expect(span.length).toBe(1);
      expect(span.get(0).props.children).toContain(message.text);
    });
    it("renders a snapshot successuly", () => {
      const tree = renderer
        .create(<Message text={message.text} type={message.type} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
