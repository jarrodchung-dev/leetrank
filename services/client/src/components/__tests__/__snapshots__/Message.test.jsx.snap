// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message danger /> renders a snapshot successuly 1`] = `
<div
  className="container is-fluid has-text-centered"
  id="notification"
>
  <div
    className="notification is-danger"
  >
    <button
      className="delete"
      onClick={[Function]}
    />
    <span>
      Danger message
    </span>
  </div>
</div>
`;

exports[`<Message success /> renders a snapshot successuly 1`] = `
<div
  className="container is-fluid has-text-centered"
  id="notification"
>
  <div
    className="notification is-success"
  >
    <button
      className="delete"
      onClick={[Function]}
    />
    <span>
      Successful message
    </span>
  </div>
</div>
`;
